src/integrationTest/java/org/opensearch/test/framework/cluster/LocalOpenSearchCluster.java

```diff
@@ -0,0 +1,491 @@
+        }
+
+        stop();
+
+        this.nodes.clear();
+        this.seedHosts = null;
+        this.initialMasterHosts = null;
+        this.clusterHomeDir = Files.createTempDirectory("local_cluster_" + clusterName + "_retry_" + retry).toFile();
+
+        start();
+    }
+
+    @SafeVarargs
+    private final Node findRunningNode(List<Node> nodes, List<Node>... moreNodes) {
+        for (Node node : nodes) {
+            if (node.isRunning()) {
+                return node;
+            }
+        }
+
+        if (moreNodes != null && moreNodes.length > 0) {
+            for (List<Node> nodesList : moreNodes) {
+                for (Node node : nodesList) {
+                    if (node.isRunning()) {
+                        return node;
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private CompletableFuture<Void> startNodes(List<NodeSettings> nodeSettingList, SortedSet<Integer> transportPorts, SortedSet<Integer> httpPorts) {
+        Iterator<Integer> transportPortIterator = transportPorts.iterator();
+        Iterator<Integer> httpPortIterator = httpPorts.iterator();
+        List<CompletableFuture<StartStage>> futures = new ArrayList<>();
+
+        for (NodeSettings nodeSettings : nodeSettingList) {
+            Node node = new Node(nodeSettings, transportPortIterator.next(), httpPortIterator.next());
+            futures.add(node.start());
+        }
+
+        return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
+    }
+
+    public void waitForCluster(ClusterHealthStatus status, TimeValue timeout, int expectedNodeCount) throws IOException {
+        Client client = clientNode().getInternalNodeClient();
+
+        
+            log.debug("waiting for cluster state {} and {} nodes", status.name(), expectedNodeCount);
+            AdminClient adminClient = client.admin();
+
+            final ClusterHealthResponse healthResponse = adminClient.cluster().prepareHealth().setWaitForStatus(status).setTimeout(timeout)
+                    .setMasterNodeTimeout(timeout).setWaitForNodes("" + expectedNodeCount).execute().actionGet();
+
+            if (log.isDebugEnabled()) {
+                log.debug("Current ClusterState:\n{}", Strings.toString(healthResponse));
+            }
+
+            if (healthResponse.isTimedOut()) {
+                throw new IOException(
+                        "cluster state is " + healthResponse.getStatus().name() + " with " + healthResponse.getNumberOfNodes() + " nodes");
+            } else {
+                log.debug("... cluster state ok {} with {} nodes", healthResponse.getStatus().name(), healthResponse.getNumberOfNodes());
+            }
+
+            assertEquals(expectedNodeCount, healthResponse.getNumberOfNodes());
+       
+    }
+
+    @Override
+    public String toString() {
+        String masterNodes = nodeByTypeToString(MASTER);
```

Comment:
Another block of master -> cluster manager

Reply:
done